### PR TITLE
Add openAPI

### PR DESCRIPTION
**Pull request by atsushii**
Addresses #9 

What I did

- Create openAPI doc for current REST APIs
- Add docker to run swagger-u by go-swagger
- Generate spec YAML by swaggo (https://github.com/swaggo/swag)

Please check the doc!

step

1, mage run
2, access to http://localhost:8002/docs

### CODE REVIEW

**Comment by aqaurius6666** [comment]
how did you generate openapi.yaml file?

**Comment by atsushii** [comment]
@aqaurius6666 
Using https://test-sso.stoplight.io/

**Comment by atsushii** [comment]
@decentralgabe 
I agree with your idea that keep these docs up to date and easy to read!
I will definitely fix my PR. Thanks for the advice!

**Comment by atsushii** [comment]
@decentralgabe 

I have fixed my code to generate spec yaml by annotations using  https://github.com/swaggo/swag

Please review it!

**Comment by decentralgabe** [comment]
thank you @atsushii this looks really good. I will take a more in depth look after the weekend.

File: cmd/main.go [hunk 1]
```diff
@@ -28,6 +29,19 @@ func init() {
...
+// @termsOfService  http://swagger.io/terms/
+
+// @contact.name   API Support
+// @contact.url    http://www.swagger.io/support
+// @contact.email  support@swagger.io
+
+// @license.name  Apache 2.0
+// @license.url   http://www.apache.org/licenses/LICENSE-2.0.html
+
+// @host      localhost:8080
```
Comment (decentralgabe) [inline cmd/main.go:44]:
```suggestion
// @title           SSI Service API
// @version         0.1
// @description     https://github.com/TBD54566975/ssi-service

// @contact.name   TBD
// @contact.url   https://github.com/TBD54566975/ssi-service/issues
// @contact.email tbd-developer@squareup.com

// @license.name  Apache 2.0
// @license.url   http://www.apache.org/licenses/LICENSE-2.0.html

// @host      localhost:8080
```

File: pkg/server/router/credential.go [hunk 2]
```diff
@@ -62,6 +63,17 @@ type CreateCredentialResponse struct {
 	Credential credsdk.VerifiableCredential `json:"credential"`
 }
 
+// ShowAccount godoc
```
Comment (decentralgabe) [inline pkg/server/router/credential.go:66]:
```suggestion
// CreateCredential godoc
```

File: pkg/server/router/credential.go [hunk 3]
```diff
@@ -87,6 +99,16 @@ type GetCredentialResponse struct {
...
```
Comment (decentralgabe) [inline pkg/server/router/credential.go:102]:
```suggestion
// GetCredential godoc
```

File: pkg/server/router/credential.go [hunk 4]
```diff
@@ -113,7 +135,19 @@ type GetCredentialsResponse struct {
 	Credentials []credsdk.VerifiableCredential `json:"credentials"`
 }
 
-// GetCredentials checks for the presence of a query parameter and calls the associated filtered get method
+// ShowAccount godoc
```
Comment (decentralgabe) [inline pkg/server/router/credential.go:138]:
```suggestion
// GetCredentials godoc
```

File: pkg/server/router/credential.go [hunk 5]
```diff
@@ -174,6 +208,17 @@ func (cr CredentialRouter) getCredentialsBySchema(schema string, ctx context.Con
 	return framework.Respond(ctx, w, resp, http.StatusOK)
 }
 
+// ShowAccount godoc
```
Comment (decentralgabe) [inline pkg/server/router/credential.go:211]:
```suggestion
// DeleteCredential godoc
```

File: pkg/server/router/did.go [hunk 6]
```diff
@@ -41,6 +42,14 @@ type GetDIDMethodsResponse struct {
 	DIDMethods []did.Method `json:"didMethods,omitempty"`
 }
 
+// ShowAccount godoc
```
Comment (decentralgabe) [inline pkg/server/router/did.go:45]:
```suggestion
// GetDIDMethods godoc
```

File: pkg/server/router/did.go [hunk 7]
```diff
@@ -41,6 +42,14 @@ type GetDIDMethodsResponse struct {
...
+// @Summary          Get DID Methods
+// @Description      get methods
```
Comment (decentralgabe) [inline pkg/server/router/did.go:47]: perhaps: Get supported DID methods?

File: pkg/server/router/did.go [hunk 8]
```diff
@@ -56,6 +65,18 @@ type CreateDIDByMethodResponse struct {
 	PrivateKey string             `json:"privateKeyBase58,omitempty"`
 }
 
+// ShowAccount godoc
```
Comment (decentralgabe) [inline pkg/server/router/did.go:68]:
```suggestion
// CreateDIDByMethod godoc
```

File: pkg/server/router/did.go [hunk 9]
```diff
@@ -91,6 +112,18 @@ type GetDIDByMethodResponse struct {
 	DID didsdk.DIDDocument `json:"did,omitempty"`
 }
 
+// ShowAccount godoc
+// @Summary          Get DID
```
Comment (decentralgabe) [inline pkg/server/router/did.go:116]:
```suggestion
// GetDIDByMethod godoc
// @Summary          Get DID
```

File: pkg/server/router/health.go [hunk 10]
```diff
@@ -14,7 +15,14 @@ const (
 	HealthOK string = "OK"
 )
 
-// Health is a simple handler that always responds with a 200 OK
+// ShowAccount godoc
```
Comment (decentralgabe) [inline pkg/server/router/health.go:18]:
```suggestion
// Health godoc
```

File: pkg/server/router/health.go [hunk 11]
```diff
@@ -14,7 +15,14 @@ const (
...
+// @Summary          Health Check
+// @Description      Health is a simple handler that always responds with a 200 OK
+// @Tags             NewSSIServer
```
Comment (decentralgabe) [inline pkg/server/router/health.go:21]: what purpose does the tag server here?

File: pkg/server/router/readiness.go [hunk 12]
```diff
@@ -23,8 +24,15 @@ type GetReadinessResponse struct {
 	ServiceStatuses map[svcframework.Type]svcframework.Status `json:"serviceStatuses"`
 }
 
-// ready runs a number of application specific checks to see if all the
-// relied upon service are healthy. Should return a 500 if not ready.
+// ShowAccount godoc
```
Comment (decentralgabe) [inline pkg/server/router/readiness.go:27]:
```suggestion
// Readiness godoc
```

File: pkg/server/router/schema.go [hunk 13]
```diff
@@ -40,6 +41,17 @@ type CreateSchemaResponse struct {
 	Schema schemalib.VCJSONSchema `json:"schema"`
 }
 
+// ShowAccount godoc
```
Comment (decentralgabe) [inline pkg/server/router/schema.go:44]:
```suggestion
// CreateSchema godoc
```

File: pkg/server/router/schema.go [hunk 14]
```diff
@@ -79,6 +100,16 @@ type GetSchemaResponse struct {
 	Schema schemalib.VCJSONSchema `json:"schema,omitempty"`
 }
 
+// ShowAccount godoc
```
Comment (decentralgabe) [inline pkg/server/router/schema.go:103]:
```suggestion
// GetSchemaByID godoc
```

**Review by decentralgabe — CHANGES_REQUESTED** [review]
minor changes - looks good!

File: docs/.swaggo [hunk 15]
```diff
@@ -0,0 +1,2 @@
+// TODO: read cryptosuite.LDKeyType from sdk properly
+skip github.com/TBD54566975/ssi-sdk/did.VerificationMethod
```
Comment (decentralgabe) [inline docs/.swaggo:2]: this should be solved before merge, what error did you run into?

File: pkg/server/router/health.go (hunk 11 above)
Comment (atsushii) [reply pkg/server/router/health.go:21]: I fix it to HealthCheck, thanks!

File: docs/.swaggo (hunk 15 above)
Comment (atsushii) [reply docs/.swaggo:2]: @decentralgabe 
`ParseComment error in file {path to did.go} :cannot find type definition: cryptosuite.LDKeyType`
the above error occurred when I ran swag init cmd.

Seems cannot load LDKeyType which define inside ssi-sdk

Comment (atsushii) [reply docs/.swaggo:2]: probably need to fix the type definition in ssi-sdk..
What do you think?

@decentralgabe

Comment (decentralgabe) [reply docs/.swaggo:2]: let me take a look. do you mind if I push to the branch if I find a solution?

**Comment by decentralgabe** [comment]
hi @atsushii I created a PR against your branch here: https://github.com/atsushii/ssi-service/pull/1

File: docs/.swaggo (hunk 15 above)
Comment (atsushii) [reply docs/.swaggo:2]: @decentralgabe 
Thanks! For sure!

**Comment by atsushii** [comment]
Hi @decentralgabe 
Thank you for your help! Already Marged your PR and fixed all of your review comments.
Please review my PR!

**Review by decentralgabe — APPROVED** [review]
awesome job, thank you for your contribution!

**Pull request merged**